GGRC-6605 Create ACR for PersonProfile

mario-amazing opened the pull request:
# Issue description
Currently all users can read person profiles through the api. So any user can get profile information of any other user. It's possible by sending GET request to <instance name>/api/people_profiles.

# Steps to test the changes

Send GET request to <instance name>/api/people_profiles. 

# Solution description

Add AccessControll rules for people_profiles api.

# Sanity checklist

- [x] I have clicked through the app to make sure my changes work and not break the app.
- [x] I have applied the correct milestone and labels.
- [x] My changes fix the issue described in the description (and do nothing else). 🤞
- [x] My changes are covered by tests.
- [x] My changes follow our [performance guidelines](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/performance.rst).
- [x] My changes follow our [js](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/javascript.rst) and/or [python](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/python.rst) guidelines.
- [x] My commits follow our [commit guidelines](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/git/how_to_write_a_commit_message.rst).

<!-- If your PR includes a migration include the additional checklist items
# Migration checklist
- [ ] Migration passes all checks from our [PR review guidelines](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/git/reviewing_pull_requests.rst#reviewing-a-pr-containing-database-migration-scripts)
- [ ] Upon merging, add 'check migration chain' and 'kokoro:force-run' labels to all open PRs with a label 'migration'
- [ ] Upon merging, update 'table structure changes' in weekly deployment documentation
-->

# PR Review checklist

- [x] The changes fix the issue and don't cause any apparent regressions.
- [x] Labels and milestone are correctly set.
- [x] The solution description matches the changes in the code.
- [x] There is no apparent way to improve the performance & design of the new code.
- [x] The pull request is opened against the correct base branch.
- [x] Upon merging, the Jira ticket's fixversion is correctly set and the ticket is moved to "QA - In Progress".

<!-- If your code is not finished yet can include a TODO check list
# TODO

- [ ] First item on the TODO
-->

olegnatsevsky commented:
@mario-amazing One more point.
I found following error trying to apply migration:
```
sqlalchemy.exc.IntegrityError: (IntegrityError) (1452, 'Cannot add or update a child row: a foreign key constraint fails (`ggrcdev`.`access_control_list`, CONSTRAINT `fk_access_control_list_base_id` FOREIGN KEY (`base_id`) REFERENCES `access_control_list` (`id`) ON DELETE CASCADE)') '\n  INSERT INTO access_control_list(\n      object_id, object_type, base_id, ac_role_id, created_at, updated_at\n  )\n  VALUES (\n      %s, %s, %s, %s, NOW(), NOW()\n  )\n  ' (27L, 'PersonProfile', 27L, 6298L)
```
Please take a look.

mario-amazing commented:
> @mario-amazing I can't find test to check described business rules.
> Like user can't read profiles of other user and so forth.
> 
> Please point me to such tests if exists or create some.

I have fixed migration. And added the test for the new logic https://github.com/google/ggrc-core/pull/9619/files#diff-3ff6e1a024e641fd3ab2d661c0527e03R64.
Could you take a look PR one more time?

olegnatsevsky commented:
@mario-amazing To track object lifecycle we have such mechanism as "Revisions". In your migration you create AccessControlList, AccessControlPerson and modify Person profile. We need to create revisions for those actions. Please check "ggrc.migrations.utils.add_to_objects_without_revisions" helper method.

The rest functionality looks good.

mario-amazing commented:
@olegnatsevsky, Added "revisions" to all created models:
https://github.com/google/ggrc-core/pull/9619/files#diff-f0069ed6f54df0f13db450293128bc97R81
https://github.com/google/ggrc-core/pull/9619/files#diff-f0069ed6f54df0f13db450293128bc97R34
https://github.com/google/ggrc-core/pull/9619/files#diff-f0069ed6f54df0f13db450293128bc97R102

victoria-kostyukova commented:
@mario-amazing please cherry-pick this commit https://github.com/google/ggrc-core/pull/9727/commits/98c093a22560fc4f2b2d0ace2cce6a32078dd89b
It was approved in #9727 pr

olegnatsevsky commented:
@mario-amazing Please rebase branch with dev and fix migration chain.

victoria-kostyukova commented:
> @mario-amazing please cherry-pick this commit [98c093a](https://github.com/google/ggrc-core/commit/98c093a22560fc4f2b2d0ace2cce6a32078dd89b)
> It was approved in #9727 pr

@mario-amazing @olegnatsevsky @IlyaTsupryk this pr needs commit mentioned above
